src/main/java/org/jenkinsci/plugins/pubsub/RunMessage.java

```diff
@@ -63,6 +67,13 @@ public RunMessage(@Nonnull Run run) {
         set(EventProps.Jenkins.jenkins_object_url, run.getUrl());
         set(EventProps.Job.job_run_queueId, Long.toString(run.getQueueId()));
 
+        // CSV list of the accumulated causes of this run
+        final List<String> causes = new ArrayList<>();
+        for (Cause cause : (List<Cause>) run.getCauses()) {
```

Comment:
Might want to wrap all of this in a check for `null` on the return of `run.getCauses()`.

Reply:
#getCauses is tagged `@Nonnull` and looks safe:

```    public @Nonnull List<Cause> getCauses() {
        CauseAction a = getAction(CauseAction.class);
        if (a==null)    return Collections.emptyList();
        return Collections.unmodifiableList(a.getCauses());
    }```

Reply:
@tfennelly ping?

Reply:
So this will be added on every event, right? I'd restrict it just the `run_started` event.